envoy/common/execution_context.h

```diff
@@ -65,4 +81,22 @@ class ScopedExecutionContext : NonCopyable {
   ExecutionContext* context_;
 };
 
+#ifdef ENVOY_ENABLE_EXECUTION_SCOPE
+#define ENVOY_EXECUTION_SCOPE_CAT_(a, b) a##b
+#define ENVOY_EXECUTION_SCOPE_CAT(a, b) ENVOY_EXECUTION_SCOPE_CAT_(a, b)
+// Invoked when |scopedObject| is active from the current line to the end of the current c++ scope.
+// |executionContext| is a pointer to the current ExecutionContext.
+// |scopedObject| is a pointer to a Envoy::Tracing::Span or a Http::FilterContext.
+#define ENVOY_EXECUTION_SCOPE(executionContext, scopedObject)                                      \
+  Envoy::Cleanup ENVOY_EXECUTION_SCOPE_CAT(on_scope_exit_, __LINE__) =                             \
```

Comment:
nit: I personally think the cat is little complex and maybe unnecessary if we use a long name? It basically won't conflict with other name.

Reply:
The cat thing is commonly used to generate unique variable names, unfortunately I'm not aware of a simpler method to achieve that.  The `ExecutionContextTest` in `test/common/common/execution_context_test.cc` currently depends on unique names in order to compile. The non-test uses currently doesn't require unique names, but I wouldn't be surprised if I need
```
ENVOY_EXECUTION_SCOPE(context,  FooObject);
....

ENVOY_EXECUTION_SCOPE(context,  BarObject);
```
in the same c++ scope in the future.

Reply:
okay. i am not familiar with macro, could you explain a little more about the `ENVOY_EXECUTION_SCOPE_CAT_` and `ENVOY_EXECUTION_SCOPE_CAT`

Reply:
Sure, https://stackoverflow.com/a/13301627 is a good explanation of this pattern.